recipes/gcc/all/conanfile.py

```diff
@@ -1,126 +1,193 @@
-        self.requires("zlib/1.2.11")
+        self.requires("gmp/6.2.1")
+        self.requires("zlib/1.2.13")
+        self.requires("isl/0.24")
 
-    def configure(self):
-        if self.settings.os == "Windows":
-            raise ConanInvalidConfiguration("Windows builds aren't supported (yet), sorry")
-        if tools.cross_building(self.settings):
-            raise ConanInvalidConfiguration("no cross-building support (yet), sorry")
+    def package_id(self):
+        del self.info.settings.compiler
 
-    def source(self):
-        tools.get(**self.conan_data["sources"][self.version])
-        extracted_dir = "gcc-%s" % self.version
-        os.rename(extracted_dir, self._source_subfolder)
+    def validate_build(self):
+        if is_msvc(self):
+            raise ConanInvalidConfiguration("GCC can't be built with MSVC")
+
+    def validate(self):
+        if self.info.settings.os == "Windows":
+            raise ConanInvalidConfiguration(
+                "Windows builds aren't currently supported. Contributions to support this are welcome."
+            )
+        if self.info.settings.os == "Macos":
+            # FIXME: This recipe should largely support Macos, however the following
+            # errors are present when building using the c3i CI:
+            # clang: error: unsupported option '-print-multi-os-directory'
+            # clang: error: no input files
+            raise ConanInvalidConfiguration(
+                "Macos builds aren't currently supported. Contributions to support this are welcome."
+            )
+        if cross_building(self):
+            raise ConanInvalidConfiguration(
+                "Cross builds are not current supported. Contributions to support this are welcome"
+            )
+
+    def layout(self):
+        basic_layout(self, src_folder="src")
+
+    def generate(self):
+        # Ensure binutils and flex are on the path.
+        # TODO: Remove when conan 2.0 is released as this will be default behaviour
+        buildenv = VirtualBuildEnv(self)
+        buildenv.generate()
+
+        tc = AutotoolsToolchain(self)
+        tc.configure_args.append("--enable-languages=c,c++,fortran")
+        tc.configure_args.append("--disable-nls")
+        tc.configure_args.append("--disable-multilib")
+        tc.configure_args.append("--disable-bootstrap")
+        tc.configure_args.append(f"--prefix={self.package_folder}")
+        tc.configure_args.append(
+            f"--libexecdir={os.path.join(self.package_folder, 'bin', 'libexec')}"
+        )
+        tc.configure_args.append(f"--with-zlib={self.deps_cpp_info['zlib'].rootpath}")
+        tc.configure_args.append(f"--with-isl={self.deps_cpp_info['isl'].rootpath}")
+        tc.configure_args.append(f"--with-gmp={self.deps_cpp_info['gmp'].rootpath}")
+        tc.configure_args.append(f"--with-mpc={self.deps_cpp_info['mpc'].rootpath}")
+        tc.configure_args.append(f"--with-mpfr={self.deps_cpp_info['mpfr'].rootpath}")
```

Comment:
```suggestion
        tc.configure_args.append(f"--with-zlib={self.dependencies['zlib'].package_folder}")
        tc.configure_args.append(f"--with-isl={self.dependencies['isl'].package_folder}")
        tc.configure_args.append(f"--with-gmp={self.dependencies['gmp'].package_folder}")
        tc.configure_args.append(f"--with-mpc={self.dependencies['mpc'].package_folder}")
        tc.configure_args.append(f"--with-mpfr={self.dependencies['mpfr'].package_folder}")
```
https://docs.conan.io/en/latest/reference/conanfile/dependencies.html#dependencies-interface

`deps_cpp_info` will be deprecated, prefer `dependencies` instead.